src/backends/meta-eis-client.c

```diff
@@ -219,34 +221,94 @@ configure_keyboard (MetaEisClient     *client,
     }
 }
 
-static void
-configure_abs (MetaEisClient     *client,
-               struct eis_device *eis_device,
-               gpointer           user_data)
+static gboolean
+has_region (struct eis_device *eis_device,
+            int                x,
+            int                y,
+            int                width,
+            int                height)
 {
-  int idx = 0;
-  cairo_rectangle_int_t rect;
-  float scale;
+  int i = 0;
 
-  if (!client->viewports)
-    return; /* FIXME: should be an error */
+  while (TRUE)
+    {
+      struct eis_region *region;
+
+      region = eis_device_get_region (eis_device, i);
+      if (!region)
+        return FALSE;
+
+      if (eis_region_get_x (region) == x &&
+          eis_region_get_y (region) == y &&
+          eis_region_get_width (region) == width &&
+          eis_region_get_height (region) == height)
+        return TRUE;
```

Comment:
wondering if it's worth checking the mapping ID matches too  and `g_warn` if not.

Reply:
It won't match, because it may happen when some user would "share" two mirrored monitors in two different streams. Useless? yes. Theoretically possible? yes. Maybe there is a more clever way to handle this, but can probably live with it for a bit.